osu.Game/Database/ArchiveModelManager.cs

```
@@ -130,56 +132,50 @@ protected ArchiveModelManager(Storage storage, IDatabaseContextFactory contextFa
         {
             notification.Progress = 0;
             notification.Text = "Import is initialising...";
 
             var term = $"{typeof(TModel).Name.Replace("Info", "").ToLower()}";
 
-            List<TModel> imported = new List<TModel>();
-
             int current = 0;
 
-            foreach (string path in paths)
+            var imported = new List<TModel>();
+
+            await Task.WhenAll(paths.Select(path => Import(path, notification.CancellationToken).ContinueWith(t =>
             {
-                if (notification.State == ProgressNotificationState.Cancelled)
-                    // user requested abort
+                if (notification.CancellationToken.IsCancellationRequested)
                     return;
 
-                try
+                lock (notification)
                 {
-                    var text = "Importing ";
-
-                    if (path.Length > 1)
-                        text += $"{++current} of {paths.Length} {term}s..";
-                    else
-                        text += $"{term}..";
-
-                    // only show the filename if it isn't a temporary one (as those look ugly).
-                    if (!path.Contains(Path.GetTempPath()))
-                        text += $"\n{Path.GetFileName(path)}";
-
-                    notification.Text = text;
-
-                    imported.Add(Import(path));
+                    current++;
 
+                    notification.Text = $"Imported {current} of {paths.Length} {term}s";
                     notification.Progress = (float)current / paths.Length;
                 }
-                catch (Exception e)
+
+                if (t.Exception == null)
+                {
+                    lock (imported)
+                        imported.Add(t.Result);
+                }
+                else
                 {
-                    e = e.InnerException ?? e;
+                    var e = t.Exception.InnerException ?? t.Exception;
```

It could have multiple inner exceptions, so this line can potentially either throw an exception on multiple inner exceptions (I don't know how the implementation works internally), or just silently ignore ("do not forward") other inner exceptions to the log. I would send the full exception to the logger. Unwrapping an exception is not the caller's responsibility.